### PR TITLE
delimiter cannot be NULL

### PR DESCRIPTION
**Pull request by MikyWoW**
Bug #2105

### CODE REVIEW

File: lib/Folder.php [hunk 1]
```diff
@@ -60,7 +60,7 @@ public function __construct(int $accountId, Horde_Imap_Client_Mailbox $mailbox,
 		$this->accountId = $accountId;
 		$this->mailbox = $mailbox;
 		$this->attributes = $attributes;
-		$this->delimiter = $delimiter;
+		$this->delimiter = $delimiter ?: '/';
```
Comment (ChristophWurst) [inline lib/Folder.php]: See https://github.com/nextcloud/mail/issues/2105#issuecomment-576693800. I'm a bit worries about the default value.

From https://tools.ietf.org/html/rfc3501#section-7.2.2

> The hierarchy delimiter is a character used to delimit levels of
>       hierarchy in a mailbox name.  A client can use it to create child
>       mailboxes, and to search higher or lower levels of naming
>       hierarchy.  All children of a top-level hierarchy node MUST use
>       the same separator character.  A NIL hierarchy delimiter means
>       that no hierarchy exists; the name is a "flat" name.

Aha! So `null` (NIL) is actually okay.

Comment (MikyWoW) [reply lib/Folder.php]: If null is valid, then this fix is probably not best solution. Maybe fix it directly on FolderMapper.php - line 219

Comment (MikyWoW) [reply lib/Folder.php]: If i fix FolderMapper.php i getting SQL error:
`An exception occurred while executing 'INSERT INTO oc_mail_mailboxes .....  SQLSTATE[HY000]: General error: 1364 Field 'delimiter' doesn't have a default value.`
DB structure need change too:
`ALTER TABLE oc_mail_mailboxes MODIFY delimiter varchar(1) DEFAULT NULL;`

Comment (ChristophWurst) [reply lib/Folder.php]: yes, we need to fix this at a few places. let's discuss in your ticket :)

**Pull request closed without merging**